model-datastructure/src/commonMain/kotlin/org/modelix/model/VersionMerger.kt

```diff
@@ -45,11 +45,32 @@ class VersionMerger(private val storeCache: IDeserializingKeyValueStore, private
         }
     }
 
+    private fun collectLatestNonMerges(version: CLVersion?, visited: MutableSet<String>, result: MutableSet<Long>) {
+        if (version == null) return
+        if (!visited.add(version.getContentHash())) return
+        if (version.isMerge()) {
+            collectLatestNonMerges(version.getMergedVersion1(), visited, result)
+            collectLatestNonMerges(version.getMergedVersion2(), visited, result)
+        } else {
+            result.add(version.id)
+            collectLatestNonMerges(version.baseVersion, visited, result)
+        }
```

Comment:
How deeply nested is this recursion usually? Is there a chance for reaching a stack overflow before this converges?